message/internal_msg_builder.go

```diff
@@ -491,6 +566,12 @@ type VMMessage struct {
 	Notification uint32 `json:"notification,omitempty"`
 }
 
+func (m *VMMessage) String() string {
+	return fmt.Sprintf(
+		"Notification: %v",
```

Comment:
`%v` is kinda the catch all - I like to be explicit:
```suggestion
		"Notification: %d",
```